File: CodeMaze-master/src/counting_bits/solution2.swift
class Solution {
    func countBits(_ num: Int) -> [Int] {
        var results = [Int]()
        
        for i in 0 ... num {
            let count = countBitsHelper(i)
            results.append(count)
        }
        
        return results
    }
    
    private func countBitsHelper(_ i: Int) -> Int {
        var count = 0
        var num = i
        while num != 0 {
            count += num & 1
            num >>= 1
        }
        return count
    }
}

File: CodeMaze-master/src/counting_bits/solution2.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> results;
        
        for (int i = 0; i <= num; ++i) {
            results.push_back(countBitsHelper(i));
        }
        
        return results;
    }
private:
    int countBitsHelper(int i) {
        int res = 0;
        while (i) {
            res += i & 1;
            i >>= 1;
        }
        
        return res;
    }
};